pkg/controller/clusterserviceversion/clusterserviceversion_controller.go

```diff
@@ -131,16 +164,65 @@ func (r *ReconcileClusterServiceVersion) Reconcile(request reconcile.Request) (r
 		return reconcile.Result{}, err
 	}
 
+	annotations := CSV.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	//check if the CSV name, matches the RHM-Operator CSV name
+	if strings.Contains(CSV.GetName(), utils.CSV_NAME) {
+		// examine DeletionTimestamp to determine if object is under deletion
+		if CSV.ObjectMeta.DeletionTimestamp.IsZero() {
+			// Case 1: the object is not being deleted
+
+			// Check if there is a finalizer attached to CSV
+			if !utils.Contains(CSV.GetFinalizers(), utils.CSV_FINALIZER) {
+				// if no -> register finalizer
+				CSV.ObjectMeta.Finalizers = append(CSV.ObjectMeta.Finalizers, utils.CSV_FINALIZER)
+				if err := r.client.Update(context.Background(), CSV); err != nil {
+					return reconcile.Result{}, err
+				}
+				reqLogger.Info("added finailzer to CSV")
+			}
+
+			err = r.checkAnnotations(CSV, annotations)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+		} else {
```

Comment:
Let's try to clean up the ifs a bit. By negating your if statement, you can move this else up and change it to just an if. Try to follow these rules when writing nested ifs and elses: https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

You can also add the golint tool locally to test your code with. Should help keep these more readable.